lsp/source/server.mts

```diff
@@ -179,6 +180,13 @@ connection.onInitialized(() => {
   }
 });
 
+const updating = async (document: { uri: string }) => {
+  const resolver = documentUpdateStatus.get(document.uri)
+  if (resolver) {
+    await resolver.promise
```

Comment:
Not sure if it's important, but you could just `return` the promise here, and avoid `async` in this function.

Then I guess it would just be `return documentUpdateStatus.get(document.uri)?.promise`.

Reply:
Indeed, this will be a bit easier.